### PR TITLE
Build testnet validator images

### PR DESCRIPTION
**Pull request by roy-dydx**
### Changelist
Build testnet validator images on push and PR to release branches. Makes testnet deploys easier.

### Test Plan
Built and tried image on on release branch.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new GitHub Actions workflow for building and pushing Docker images to AWS ECR for the testnet protocol.
	- Added several scripts to streamline the setup and management of testnet environments, including automation for starting nodes and creating snapshots.

- **Bug Fixes**
	- Enhanced error handling in script execution to improve reliability during node management tasks.

- **Documentation**
	- Added comments and usage instructions in scripts to facilitate user understanding and ease of use.

### CODE REVIEW

File: protocol/testing/testnet/cosmovisor/genesis/bin/dydxprotocold
Comment (teddyding) [inline protocol/testing/testnet/cosmovisor/genesis/bin/dydxprotocold:1]: Which version is this for my understanding? Does this matter?

File: protocol/testing/testnet/testnet.sh [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+# This file initializes the home directories of validators / full-nodes for validators run for
+# the private test-net on AWS.
+
+source "./vars.sh"
+
+CHAIN_ID="dydx-testnet-1"
+
+# Define the mapping from version to URL
+declare -A version_to_url
+version_to_url["v5.1.0"]="https://github.com/dydxprotocol/v4-chain/releases/download/protocol%2Fv5.1.1/dydxprotocold-v5.1.1-linux-amd64.tar.gz"
```
Comment (teddyding) [inline protocol/testing/testnet/testnet.sh:13]: Should we add `v5.2.0` here?

File: protocol/testing/testnet/start.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# Set up CosmosVisor.
+for i in "${!MONIKERS[@]}"; do
+	VAL_HOME_DIR="$HOME/chain/.${MONIKERS[$i]}"
+	# Copy binaries for `cosmovisor` from the docker image into the home directory.
+	# Work-around to ensure docker volume contains the same binaries as the git repo.
+	cp -r "$HOME/cosmovisor" "$VAL_HOME_DIR/"
+done
+
+# Set up CosmosVisor for full-nodes.
+for i in $(seq 0 $LAST_FULL_NODE_INDEX); do
```
Comment (teddyding) [inline protocol/testing/testnet/start.sh:20]: Why is this 0 through 2? I thought there are only 2 full nodes?

Comment (roy-dydx) [reply protocol/testing/testnet/start.sh:20]: No harm in initializing more home directories

File: protocol/testing/testnet/testnet.sh (hunk 1 above)
Comment (roy-dydx) [reply protocol/testing/testnet/testnet.sh:13]: Yeah we should specify previous version. This is getting backported to v5.2.0 branch first though. Will follow up with a PR to set everything for v6.0.0

File: protocol/testing/testnet/cosmovisor/genesis/bin/dydxprotocold
Comment (roy-dydx) [reply protocol/testing/testnet/cosmovisor/genesis/bin/dydxprotocold:1]: genesis (v1.0.0)

**Review by teddyding — APPROVED** [review]

**Comment by roy-dydx** [comment]
@Mergifyio backport release/protocol/v5.2.x

**Comment by roy-dydx** [comment]
https://github.com/Mergifyio backport release/protocol/v6.x

**Pull request merged**